framework/logging.cpp

```diff
@@ -1684,11 +1777,29 @@ void KeyValuePairLogger::prepare_line_prefix()
     timestamp_prefix += test->id;
 }
 
-void KeyValuePairLogger::print(int tc)
+void KeyValuePairLogger::print(int tc, ChildExitStatus status)
 {
+    std::string skip_message = "skip";
+    std::string skip_message_from_main_thread;
+    get_skip_message(-1, skip_message_from_main_thread);
+
     logging_printf(LOG_LEVEL_QUIET, "%s_result = %s\n", test->id,
-                   state == TestSkipped ? "skip" :
+                   state == TestSkipped ? skip_message.c_str() :
                    state == TestFailed ? "fail" : "pass");
+    
+    if(status.result == TestPassed && state == TestSkipped){ // if test passed in init and skipped on all threads in run
+        logging_printf(LOG_LEVEL_QUIET, "%s_skip_category = %s\n", test->id, "RUNTIME_SKIP");
+        logging_printf(LOG_LEVEL_QUIET, "%s_skip_reason = %s\n", test->id, "Skipped in Run. Check individual thread messages");
+    }
+    else{
+        if(skip_message_from_main_thread.size() > 0){
```

Comment:
This is misleading name and concept. A test skips on "main thread" when it skips in the `init()` function which is executed single-threaded before threads are launched and execute `run()`. So this should be called `init_skip_message`. The entire concept of "main thread" should be removed from the code.

Reply:
This part collects message from thread0. Thread0 can either be the "main" thread in `init` function or `cpu=0` in `run` function. Skip can happen in `init` or all cpus in `run` function can skip. In both these cases, we are getting the skip message from thread 0. Thread 0 is called main thread in the `init` function where as just `Thread: 0` in the `run`. perhaps i should change to `skip_message_from_thread0`?

I feel the `init_skip_message` doesn't capture the essence of all thread skips in the `run` function.

Reply:
this actually code treats only init skips. but alright, you're getting it using `get_skip_message(-1, ...)` somewhere else. so it just should be called `skip_message` -- it's the entire tests "skip message".

Reply:
Sorry, i was wrong. Just jumped to the conclusion too early without much understanding. You are right it's only skip message from `init` function. Let me change it `init_skip_message` itself.